### PR TITLE
Fix scope lookups in image and cluster Vulnerabilities sub-resolvers

### PR DESCRIPTION
**Pull request by charmik-redhat**
## Description

Some image and cluster vulnerability sub resolvers used to search the default ctx sent to the sub-resolver. The default ctx would not have the entity scoping added when vulnerabilities were queried from another entity. This PR updates the sub-resolvers in image and cluster vulnerability sub-resolvers to use the parent ctx to lookup scopes.

## Checklist
- [x] Investigated and inspected CI test results
- ~[ ] Unit test and regression tests added~
- ~[ ] Evaluated and added CHANGELOG entry if required~
- ~[ ] Determined and documented upgrade steps~
- ~[ ] Documented user facing changes (create PR based on [openshift/openshift-docs](https://github.com/openshift/openshift-docs) and merge into [rhacs-docs](https://github.com/openshift/openshift-docs/tree/rhacs-docs))~

If any of these don't apply, please comment below.

## Testing Performed

Manual testing 
- Check if `fixedByVersion`, `isFixable` and `discoveredAtImage` in ImageVulnerability graphQL work as expected when `imageVulnerabilities` are queries from other entities.
- Check if `fixedByVersion` and `isFixable` in ClusterVulnerability graphQl work as expected.

### CODE REVIEW

**Review by keyallis — APPROVED** [review]
Good catch on these, they were probably changes I made when migrating old code over

File: central/graphql/resolvers/cluster_vulnerabilities.go [hunk 1]
```diff
@@ -386,7 +388,7 @@ func (resolver *clusterCVEResolver) EnvImpact(ctx context.Context) (float64, err
 
 func (resolver *clusterCVEResolver) FixedByVersion(ctx context.Context) (string, error) {
```
Comment (md2119) [inline central/graphql/resolvers/cluster_vulnerabilities.go:389]: nit: `_ context.Context`

File: central/graphql/resolvers/cluster_vulnerabilities.go [hunk 2]
```diff
@@ -386,7 +388,7 @@ func (resolver *clusterCVEResolver) EnvImpact(ctx context.Context) (float64, err
...
 	defer metrics.SetGraphQLOperationDurationTime(time.Now(), pkgMetrics.ClusterCVEs, "FixedByVersion")
-	scope, hasScope := scoped.GetScope(ctx)
+	scope, hasScope := scoped.GetScope(resolver.ctx)
```
Comment (md2119) [inline central/graphql/resolvers/cluster_vulnerabilities.go:391]: pass the `resolver.ctx` to the datastore

File: central/graphql/resolvers/cluster_vulnerabilities.go [hunk 3]
```diff
@@ -414,7 +416,7 @@ func (resolver *clusterCVEResolver) IsFixable(ctx context.Context, args RawQuery
 	conjuncts := []*v1.Query{query, search.NewQueryBuilder().AddBools(search.Fixable, true).ProtoQuery()}
 
 	// check scoping, add as conjunction if needed
-	if scope, ok := scoped.GetScope(ctx); !ok || scope.Level != v1.SearchCategory_CLUSTER_VULNERABILITIES {
+	if scope, ok := scoped.GetScope(resolver.ctx); !ok || scope.Level != v1.SearchCategory_CLUSTER_VULNERABILITIES {
```
Comment (md2119) [inline central/graphql/resolvers/cluster_vulnerabilities.go:419]: pass the `resolver.ctx` to the datastore

**Review by md2119 — CHANGES_REQUESTED** [review]
Although conjunction query are built, make sure datastores are getting the correct context to reduce the confusion.

**Review by md2119 — APPROVED** [review]

**Pull request merged**